### PR TITLE
CRM-17602 - Check Logic when searching with Date Range

### PR DESCRIPTION
**Pull request by saurabhbatra96**
- [CRM-17602: Check Logic when searching with Date Range](https://issues.civicrm.org/jira/browse/CRM-17602)

### CODE REVIEW

File: CRM/Contribute/BAO/Query.php [hunk 1]
```diff
@@ -1197,4 +1198,30 @@ public static function buildDateWhere(&$values, $query, $name, $field, $title) {
...
+    }
+    $lowDate = strtotime($fields['contribution_date_low']);
+    $highDate = strtotime($fields['contribution_date_high']);
+
+    if ($lowDate > $highDate) {
+      $errors['contribution_date'] = ts('Please check your Contribution Date Range.');
+      // remove this after errors starts working.
+      echo ts('Please check your Contribution Date Range.');
+    }
+    return empty($errors) ? TRUE : $errors;
```
Comment (saurabhbatra96) [inline CRM/Contribute/BAO/Query.php]: The function works properly (checked with the test echo statement) however the errors do not show up.

**Comment by saurabhbatra96** [comment]
re test this please

**Comment by saurabhbatra96** [comment]
re test this

**Comment by kurund** [comment]
jenkins, test this please

**Comment by mlutfy** [comment]
jenkins, test this please

**Comment by mlutfy** [comment]
output from the previous build seems OK, but jenkins didn't report back?
https://test.civicrm.org/job/CiviCRM-Core-PR/8214/console

**Comment by mlutfy** [comment]
Quick summary of IRC conversation:

- if possible, remove code duplication for the formRule() checks, and have one common function
- this will also help with removing the number of new ts() strings, ex: `ts('%1: Please check that your date range is in correct chronological order.', array(1 => '?')`
- per civicrm convention, might be better to have the validation rule logic in CRM/Utils/Type.php, and QF code in CRM/Core/Form.php? (avoid mixing BAO & QF code when possible, although admittedly, sometimes hard to avoid, such as in the Query classes)

**Comment by saurabhbatra96** [comment]
@mlutfy Have a look when you have a minute or two.

**Pull request merged**